ndk/src/system_properties.rs

```diff
@@ -0,0 +1,326 @@
+            Ok(CString::from_vec_with_nul(value)?)
+        }
+    }
+}
+
+/// Returns the property value as an owned [`CString`] with possibly invalid UTF-8 [but no interor
+/// NULs].  The maximum length can be up to 92 ([`ffi::PROP_VALUE_MAX`]) including NUL terminator.
+///
+/// [but no interor NULs]: GetRawError::NulError
+///
+/// See [`get()`] for a more convenient API that validates this string for UTF-8 and directly parses
+/// it into a [`FromStr`]-compatible type.
+///
+/// # Deprecation
+/// Deprecated since Android O (API level 26), use [`Property::find()`] with
+/// [`Property::read_callback()`] instead which does not have a limit on `value` nor `name` length.
+#[doc(alias = "__system_property_get")]
+pub fn get_raw(name: &CStr) -> Result<CString, GetRawError> {
+    process_owned(|value| unsafe { ffi::__system_property_get(name.as_ptr(), value) })
+}
+
+/// Possible failures returned by [`get()`].
+#[allow(missing_debug_implementations)] // Our MSRV is too low for derive(Debug) to emit bounds on T.
+#[derive(Error)]
+pub enum GetError<T> {
+    #[error("Property is missing or empty")]
+    MissingOrEmpty,
+    #[error(transparent)]
+    NulError(#[from] FromBytesWithNulError),
+    #[error("Property does not contain valid UTF-8")]
+    Utf8Error(#[from] Utf8Error),
+    #[error(transparent)]
+    Io(#[from] Error),
+    #[error(transparent)]
+    ParseError(T),
+}
+
+/// Internal helper to deduplicate the implementation between [`get()`] and [`Property::read()`].
+fn process_parse<T: FromStr>(get: impl FnOnce(*mut c_char) -> i32) -> Result<T, GetError<T::Err>> {
+    let mut value = [0u8; ffi::PROP_VALUE_MAX as usize];
+    let ret = get(value.as_mut_ptr());
+    match ret {
+        0 => Err(GetError::MissingOrEmpty),
+        ..=-1 => Err(Error::from_raw_os_error(-ret).into()),
+        1.. => {
+            let c_str = CStr::from_bytes_with_nul(&value[..ret as usize + 1])?;
+            c_str.to_str()?.parse().map_err(GetError::ParseError)
+        }
+    }
+}
+
+/// Returns the property value as a [`FromStr`]-parsed type from a source string of at most 92
+/// ([`ffi::PROP_VALUE_MAX`]) characters, including NUL terminator.
+///
+/// # Implementation details
+/// This is implemented without any up-front allocations like [`get_raw()`], but requires a trip
+/// through [`CStr`] and [`str`] (for calling [`FromStr::from_str()`]) meaning the resulting
+/// string has to be compliant with [`CStr`] ([no interior NULs]) and [`str`] ([valid UTF-8]).  In
+/// other words, parsing into a [`String`] will never contain interior NULs (and it is unknown and
+/// unlikely whether the property API allows for this).
+///
+/// [no interior NULs]: GetError::NulError
+/// [valid UTF-8]: GetError::Utf8Error
+///
+/// # Deprecation
+/// Deprecated since Android O (API level 26), use [`Property::find()`] with
+/// [`Property::read_callback()`] instead which does not have a limit on `value` nor `name` length.
```

Comment:
Should the docs here maybe also mention the limit on what the limit on returned value is, like is done for the property name?

Reply:
That's already covered in the first paragraph / "`Returns xxx` title" of this function?

Perhaps I should remove `` and `name` `` here as we've never introduced a limit on `name` length in this function and as demonstrated elsewhere, there is none for `get()` (only for `__system_property_read()`).